### PR TITLE
#[UIC-1419]:added new endpoint to create database connection and tables in superset db 

### PR DESCRIPTION
**Pull request by jitendra-kumawat**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
**Requirement**: 
Expose a single API from superset to dump db connection, tables, visualizations and dashboard metadata in superset db in respective  tables with default info and minimum required info(take as payload)

**Assumptions:**
1.provided minimum info is correct ,so that while viewing dashboard run time errors can be avoided

**Behaviour:**
1.This API  will update superset db with default info and provide info without and connection validation
2. If there is any runtime errors in  loading dashboard then user will manually update configurations as per superset work flow
3.List of columns and metrics are not available in newly created table, to update columns in table we need to manually re-save table connection. or re-sync table columns from superset UI
4. There is no modification in Table column is done at the time of saving metadata like Columns Data should be in standard SQL data format .
5.For LineTimeSeries chart DateTime column data should be in DateTime format ,to support another format like epoch we need to manually update configuration  (follow  step-3 )

**API Details**
API  :  <server> /superset/add_to_dashboard
Method  : POST
Payload  type :form-data 
Sample payload- shared in mail/channel

Required info in request to setup connection :--
 csrf_token
 cookies

Screenshot 2019-06-12 at 5 14 24 PM

### TEST PLAN
Validation done using Postman 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@bipinsoniguavus 
@ankursinghal2005 
@rasmi-ranjan-guavus

### CODE REVIEW

File: superset/connectors/sqla/views.py [hunk 1]
```diff
@@ -18,19 +18,19 @@
...
-from flask import flash, Markup, redirect
+from flask import flash, Markup, redirect,request, Response
 from flask_appbuilder import CompactCRUDMixin, expose
 from flask_appbuilder.actions import action
 from flask_appbuilder.models.sqla.interface import SQLAInterface
 from flask_appbuilder.security.decorators import has_access
 from flask_babel import gettext as __
 from flask_babel import lazy_gettext as _
-
+import simplejson as json
```
Comment (bipinsoniguavus) [inline superset/connectors/sqla/views.py]: Seems like a redundant import, please remove it

File: superset/connectors/sqla/views.py [hunk 2]
```diff
@@ -269,6 +269,29 @@ class TableModelView(DatasourceModelView, DeleteMixin, YamlExportMixin):  # noqa
...
+            db.session.add(table_model)
+            db.session.commit()
+            logging.info('table is created with id = '+str(table_model.id)+' and linked with database id = '+str(database_id))
+        except Exception as e:
+            logging.exception(e)
+            return json_error_response(e)
+
+        return json_success(json.dumps({'table_name': table_model.name}))
+
+
```
Comment (bipinsoniguavus) [inline superset/connectors/sqla/views.py]: Remove extra lines added

File: superset/connectors/sqla/views.py (hunk 1 above)
Comment (jitendra-kumawat) [reply superset/connectors/sqla/views.py]: used in json_success(json.dumps({'table_name': table_model.name}))

File: superset/connectors/sqla/views.py (hunk 2 above)
Comment (jitendra-kumawat) [reply superset/connectors/sqla/views.py]: done

**Pull request merged**